models/service.go

```diff
@@ -52,11 +52,20 @@ func GetServiceDetails(namespaceName, serviceName string) (*Service, error) {
 	service := Service{}
 	service.Name = serviceName
 	service.Namespace = Namespace{namespaceName}
-	service.setKubernetesDetails(istioClient)
-	service.setIstioDetails(istioClient)
-	service.setPrometheusDetails()
 
-	return &service, nil
+	if err = service.setKubernetesDetails(istioClient); err != nil {
```

Comment:
It is idiomatic Go. If you were to omit the `err != nil`, the compiler will tell you.

Reply:
@pilhuhn This is a typical construction from GoLang. I think it is going to be more scenarios like this in the app in the future since it is pretty handy. Actually, I found 15+ occurrences in our current code base.